### PR TITLE
Shallow clone if --since-commit is provided

### PR DESCRIPTION
**Pull request by mcastorina**

### CODE REVIEW

File: pkg/sources/git/git.go [hunk 1]
```diff
@@ -472,7 +475,71 @@ func TryAdditionalBaseRefs(repo *git.Repository, base string) (*plumbing.Hash, e
...
+	uri, err := url.Parse(uriString)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to parse Git URI: %s", err)
+	}
+
+	if uri.Scheme == "file" || uri.Host != "github.com" {
+		return PrepareRepo(uriString)
+	}
+
+	owner, repoName, found := strings.Cut(uri.Path, "/")
```
Comment (mcastorina) [inline pkg/sources/git/git.go:499]: Oops.. looks like `strings.Cut` was added in go1.18

Comment (dustin-decker) [reply pkg/sources/git/git.go:499]: #566 upgrades to 1.18 for CI and builds

**Comment by dustin-decker** [comment]
Do you think the test failure might be related?

**Comment by mcastorina** [comment]
It could be? The failure *is* related to cloning, but the tests pass locally for me. I'm rerunning the action now.

**Comment by mcastorina** [comment]
Still failing :/

File: pkg/sources/git/git.go [hunk 2]
```diff
@@ -472,7 +475,71 @@ func TryAdditionalBaseRefs(repo *git.Repository, base string) (*plumbing.Hash, e
...
+	var ts string
+	{
+		author := commit.GetAuthor()
+		if author == nil {
+			return PrepareRepo(uriString)
+		}
+		ts = author.GetDate().Format(time.RFC3339)
+	}
+
+	remotePath := fmt.Sprintf("%s://%s%s", uri.Scheme, uri.Host, uri.Path)
```
Comment (ahrav) [inline pkg/sources/git/git.go:518]: are you missing a trailing `/` after the second `%s`

Comment (mcastorina) [reply pkg/sources/git/git.go:518]: That's how the original code I copied this from does it, and I believe `uri.Path` begins with a leading `/`

File: pkg/sources/git/git.go [hunk 3]
```diff
@@ -216,9 +217,11 @@ func CloneRepo(userInfo *url.Userinfo, gitUrl string) (clonePath string, repo *g
 		err = errors.WrapPrefix(err, "could not parse url", 0)
 		return
 	}
-
 	cloneURL.User = userInfo
-	cloneCmd := exec.Command("git", "clone", cloneURL.String(), clonePath)
+
+	gitArgs := []string{"clone", cloneURL.String(), clonePath}
```
Comment (mcastorina) [inline pkg/sources/git/git.go:222]: Found the issue - I accidentally constructed `gitArgs` before setting `cloneURL.User` here (fixed in latest commit). Strange that it didn't happen locally, but I'm glad the CI caught it :)

File: pkg/sources/git/git.go [hunk 4]
```diff
@@ -472,7 +475,71 @@ func TryAdditionalBaseRefs(repo *git.Repository, base string) (*plumbing.Hash, e
...
+	if !found {
+		return PrepareRepo(uriString)
+	}
+
+	client := github.NewClient(nil)
+	commit, _, err := client.Git.GetCommit(context.Background(), owner, repoName, commitHash)
+	if err != nil {
+		return PrepareRepo(uriString)
+	}
+	var ts string
```
Comment (bill-rich) [inline pkg/sources/git/git.go:509]: nit: `ts` could be changed to something longer like `timestamp` so it's easier to understand what it is.

File: pkg/sources/git/git.go (hunk 2 above)
Comment (bill-rich) [inline pkg/sources/git/git.go:518]: Since you already have the url parsed, you could use `uri.String()` instead of reassembling it manually.

Comment (mcastorina) [reply pkg/sources/git/git.go:518]: Ah yeah, good point!

**Pull request merged**